standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnDbUtil.java

```diff
@@ -385,6 +391,26 @@ public static String queryToString(Configuration conf, String query, boolean inc
     return sb.toString();
   }
 
+  /**
+   * This is only for testing, it does not use the connectionPool from TxnHandler!
+   * @param conf
+   * @param query
+   * @throws Exception
+   */
+  @VisibleForTesting
+  public static void executeUpdate(Configuration conf, String query)
```

Comment:
That's not a test class. Production code becomes massive because of that.

Reply:
Well, this class is a test utility.
/**
 * Utility methods for creating and destroying txn database/schema, plus methods for
 * querying against metastore tables.
 * Placed here in a separate class so it can be shared across unit tests.
 */
public final class TxnDbUtil

The problem is more like getEpochFn and executeQueriesInBatchNoCount was added to this class, those are production code. I know it would be nicer if it would be in a test package, but then it would be harder to use in 5 different projects

Reply:
in this case we should consider refactoring this class

Reply:
Created a Jira for that: https://issues.apache.org/jira/browse/HIVE-24477